lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +988,57 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
+        if module.check_mode:
+            result.update(changed=True)
+            module.exit_json(**result)
+
+        # Archive already exists, then fail
+        if os.path.exists(archive):
+            module.fail_json(msg="Archive file %s already exists" % archive,
```

Comment:
@willthames yep, thats the only way I see of doing that, create in tmp dir, md5sum both of them, if match OK, if different CHANGED. Although that does add some complexity.